testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -0,0 +1,511 @@
+
+    # Replicating docs to SG so that later we can create delete event
+    # Adding Listener for replicator
+    replicator = Replication(base_url)
+    repl_config = replicator.configure(source_db=cbl_db,
+                                       target_url=sg_blip_url,
+                                       continuous=False,
+                                       headers=session_header,
+                                       replication_type="push")
+    repl = replicator.create(repl_config)
+    repl_change_listener = replicator.addReplicatorEventChangeListener(repl)
+
+    # 2. Starting Replication and waiting for it finish
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # 3 deleting some docs, so that we get delete event
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)["rows"]
+    docs_to_modify = random.sample(sg_docs, (num_of_docs / 10) * 2)
+    for doc in docs_to_modify:
+        doc_id = doc["id"]
+        sg_client.delete_doc(url=sg_url, db=sg_db, doc_id=doc_id, rev=doc["value"]["rev"], auth=auth_session)
+
+    # 4 Replicating for deleted docs
+    repl_delete_config = replicator.configure(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=False,
+                                              headers=session_header,
+                                              replication_type="pull")
+    repl_delete = replicator.create(repl_delete_config)
+    repl_delete_change_listener = replicator.addReplicatorEventChangeListener(repl_delete)
+    replicator.start(repl_delete)
+    replicator.wait_until_replicator_idle(repl_delete)
+
+    # 5. Verifying the delete event in event captures
+    doc_delete_event_changes = replicator.getReplicatorEventChanges(repl_delete_change_listener)
+    replicator.removeReplicatorEventListener(repl_delete, repl_delete_change_listener)
+    replicator.stop(repl_delete)
+    event_dict = get_event_changes(doc_delete_event_changes)
+    replicator.removeReplicatorEventListener(repl_delete, repl_delete_change_listener)
+    replicator.stop(repl_delete)
+    assert len(event_dict) != 0, "Replication listener didn't caught events. Check app logs for detailed info"
+    for doc_id in event_dict:
+        assert event_dict[doc_id]["flags"] == "1" or event_dict[doc_id]["flags"] == "[DocumentFlagsDeleted]" or \
+            event_dict[doc_id]["flags"] == "Deleted", \
+            'Deleted flag is not tagged for document. Flag value: {}'.format(event_dict[doc_id]["flags"])
+
+    # Verifying if the docs, for which access has be revoked, are purged
+    doc_ids = db.getDocIds(cbl_db)
+    for sg_doc in docs_to_modify:
+        assert sg_doc["id"] not in doc_ids, "channel access removal didn't purge the docs from cbl db"
```

Comment:
as it is not channel access, can you modify assert stmt according to the test?

Reply:
changed to "Revoked access to channel didn't purge the docs from cbl db"